copilot/ask_websocket.py

```diff
@@ -0,0 +1,28 @@
+from pieces_os_client import QGPTStreamOutput,QGPTStreamInput
+from websocket import WebSocketConnectionClosedException
+
+from ..settings import PiecesSettings
+from ..base_websocket import BaseWebsocket
+
+class AskStreamWS(BaseWebsocket):
+	def __new__(cls,*args,**kwargs):
+		if not hasattr(cls, 'instance'):
+			cls.instance = super(AskStreamWS, cls).__new__(cls)
+		return cls.instance
+
+	@property
+	def url(self):
+		return PiecesSettings.ASK_STREAM_WS_URL
+
+	def on_message(self,ws, message):
+		self.on_message_callback(QGPTStreamOutput.from_json(message))
+
+	
+	def send_message(self,message:QGPTStreamInput):
+		try:
+			if not self.ws:
+				raise WebSocketConnectionClosedException()
+			self.ws.send(message.to_json())
+		except WebSocketConnectionClosedException:
```

Comment:
what does the error handling look like here when this fails as a user?

Reply:
nothing happens here it just reconnect to the websocket if it is not